lfs/localstorage.go

```diff
@@ -0,0 +1,46 @@
+package lfs
+
+import (
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/github/git-lfs/vendor/_nuts/github.com/rubyist/tracerx"
+)
+
+func ClearTempObjects() {
+	filepath.Walk(LocalObjectTempDir, func(path string, info os.FileInfo, err error) error {
+		if shouldDeleteTempObject(path, info) {
+			return os.RemoveAll(path)
+		}
+
+		return err
+	})
+}
+
+func shouldDeleteTempObject(path string, info os.FileInfo) bool {
+	if info == nil || info.IsDir() {
+		return false
+	}
+
+	base := filepath.Base(path)
+	parts := strings.SplitN(base, "-", 2)
+	oid := parts[0]
+	if len(parts) < 2 || len(oid) != 64 {
+		tracerx.Printf("Removing invalid tmp object file: %s", path)
+		return true
+	}
+
+	if FileExists(localMediaPathNoCreate(oid)) {
+		tracerx.Printf("Removing existing tmp object file: %s", path)
+		return true
+	}
+
+	if time.Since(info.ModTime()) > time.Hour {
```

Comment:
I tried that approach, but I don't think my locks were doing what I thought they were. The contentaddressable package opens them with these flags: `os.O_WRONLY|os.O_CREATE|os.O_EXCL`. Turns out `O_EXCL` + `O_CREATE` only confirm the file didn't exist previously. `os.RemoveAll` happily deleted it.

Cross platform locking seemed too complicated for v1.0.1, so I went with this workaround. Considering that temp files will be removed most of the time, it seems like a fair tradeoff.